bashminify.sh

```diff
@@ -3,26 +3,59 @@
 # Bashminify
 
 # minify bash scripts
-# Pascal Sommer, 2016
-
-set -u
 
 function print_error {
-    echo -e "\033[0;31mERROR\033[0m:" $@ >&2
+	echo -e "\033[0;31mERROR\033[0m:" "$@" >&2
 }
 
-source read_args.sh
+input_file=$1
+if [[ -z "$input_file" ]]; then
+	print_error "Expected 1 argument"
```

Comment:
This here seems like it could confuse users because if the user passes an argument but the argument is not a filename then the error "Expected 1 argument" appears, but instead there should probably be an error message telling the user that the file doesn't exist. Before this if statement, we could add another if statement that checks if 1 argument was passed.

Also, the script should probably exit with a non-zero error code at this point, currently the script continues with the invalid file name.

Reply:
I've added a check for read permissions and the necessary exit calls.

I did not check for the file type, however, as it gets a little more complex when symlinks are involved.

Reply:
oh right, I didn't think about symlinks, good that you noticed that